action/action_test.go

```diff
@@ -0,0 +1,238 @@
+package action
```

Comment:
Now that we are effectively promoting these packages for reuse by others, I'd prefer black-box testing from the `action_test` package. Benefits include:
* the testcases are more likely to document the expected use of the package
* any test-specific externals of the package will stick out like a sore thumb, thus tending to reduce such externals.